docker-compose.yml

```diff
@@ -25,3 +28,29 @@ services:
     image: "redis:4.0.9"
   zookeeper:
     image: "zookeeper:3.4.10"
+  redis-cluster-node-0:
```

Comment:
wow, this is a lot of stuff. do we actually need a three node cluster to test this out? or can we mock it more simply?

Reply:
One idea would be to use
```
grokzen/redis-cluster
```
which gives us a Redis cluster with a single Docker dependency. It is not recommended for production use but perfect for testing environments. I will say, Redis cluster has idiosyncrasies that are worth having tests run against a real instance but that depends entirely on the tests.

Reply:
we're not actually implementing a full redis cluster client here, thankfully, but just wrapping an existing one. do you think we need to have behavioral tests here in baseplate beyond ensuring that we generate spans?

Reply:
There were a few cases where I was thinking more testing would be useful but ultimately any test I could think of would just end up testing the underlying `redis-py-cluster` library, so didn't add anything beyond what the redis client already has (I did end up using the `redis-cluster` container for the integration tests though, it was also pretty handy for manual testing)